plugins/inputs/ping/ping.go

```diff
@@ -157,204 +141,104 @@ func (p *Ping) Gather(acc telegraf.Accumulator) error {
 	return nil
 }
 
-func getAddr(iface string) string {
-	if addr := net.ParseIP(iface); addr != nil {
-		return addr.String()
-	}
-
-	ifaces, err := net.Interfaces()
-	if err != nil {
-		return ""
-	}
-
-	var ip net.IP
-	for i := range ifaces {
-		if ifaces[i].Name == iface {
-			addrs, err := ifaces[i].Addrs()
-			if err != nil {
-				return ""
-			}
-			if len(addrs) > 0 {
-				switch v := addrs[0].(type) {
-				case *net.IPNet:
-					ip = v.IP
-				case *net.IPAddr:
-					ip = v.IP
-				}
-				if len(ip) == 0 {
-					return ""
-				}
-				return ip.String()
-			}
-		}
-	}
-
-	return ""
-}
-
-func hostPinger(binary string, timeout float64, args ...string) (string, error) {
-	bin, err := exec.LookPath(binary)
+func (p *Ping) pingToURLNative(destination string, acc telegraf.Accumulator) {
+	pinger, err := ping.NewPinger(destination)
 	if err != nil {
-		return "", err
+		p.Log.Errorf("Failed to ping: %v", err)
+		acc.AddError(err)
+		return
 	}
-	c := exec.Command(bin, args...)
-	out, err := internal.CombinedOutputTimeout(c,
-		time.Second*time.Duration(timeout+5))
-	return string(out), err
-}
 
-func filterIPs(addrs []net.IPAddr, filterFunc IsCorrectNetwork) []net.IPAddr {
-	n := 0
-	for _, x := range addrs {
-		if filterFunc(x) {
-			addrs[n] = x
-			n++
-		}
+	// Required for windows. Despite the method name, this should work without the need to elevate privileges and has been tested on Windows 10
+	if runtime.GOOS == "windows" {
+		pinger.SetPrivileged(true)
 	}
-	return addrs[:n]
-}
-
-func hostResolver(ctx context.Context, ipv6 bool, destination string) (*net.IPAddr, error) {
-	resolver := &net.Resolver{}
-	ips, err := resolver.LookupIPAddr(ctx, destination)
 
-	if err != nil {
-		return nil, err
+	if p.IPv6 == true {
+		pinger.SetNetwork("ip6")
 	}
 
-	if ipv6 {
-		ips = filterIPs(ips, isV6)
+	// The interval cannot be below 0.2 seconds, matching ping implementation: https://linux.die.net/man/8/ping
+	if p.PingInterval < 0.2 {
+		pinger.Interval = time.Duration(.2 * float64(time.Second))
 	} else {
-		ips = filterIPs(ips, isV4)
-	}
-
-	if len(ips) == 0 {
-		return nil, errors.New("Cannot resolve ip address")
+		pinger.Interval = time.Duration(p.PingInterval * float64(time.Second))
 	}
```

Comment:
Can you please do this in the `Init()` function.

Reply:
So to get pinger with `pinger, err := ping.NewPinger(destination)` it resolves the destination address. I assumed we want to re-do this at each gather in case something changes? What do you think?

I would have to move ping.NewPinger to Init() as well to move setting the Interval and Timeout to Init().

Reply:
No I was think about clipping `p.PingInterval` in `Init()` and get rid of the `if p.PingInterval < 0.2 {` here.